strawberryfields/api/devicespec.py

```diff
@@ -151,6 +151,7 @@ def create_program(self, **parameters):
         # evaluate the blackbird template
         bb = bb(**parameters)
         prog = sf.io.to_program(bb)
+        prog._compiled_with = self.default_compiler
```

Comment:
I would potentially have a tuple here:

```python
prog._compile_info = (self, self.default_compiler)
```

e.g., the program stores both the device spec used to compile it, as well as the compiler used.